webbpsf/trending.py

```diff
@@ -628,18 +638,44 @@ def single_measurement_trending_plot(
         print('     Previous OPD is:', prev_filename)
 
     prev_opd, prev_opd_hdu = _read_opd(prev_filename)
+    prev_sensing_apername = prev_opd_hdu[0].header['APERNAME']
 
     if subtract_target:
         if verbose:
             print('     Subtracting NIRCam SI WFE target phase map')
 
-        # Get WSS Target Phase Map
-        was_targ_file = os.path.join(webbpsf.utils.get_webbpsf_data_path(), 'NIRCam', 'OPD', 'wss_target_phase_fp1.fits')
+        # Get WSS Target Phase Map for the sensing aperture
+        # Note that the sensing maintenance program changed field point from NRC A3 to A1 around Dec 2024.
+        if sensing_apername == 'NRCA3_FP1':
+            was_targ_file = os.path.join(
+                webbpsf.utils.get_webbpsf_data_path(), 'NIRCam', 'OPD', 'wss_target_phase_fp1.fits'
+            )
+        elif sensing_apername == 'NRCA1_FP6':
+            was_targ_file = os.path.join(
+                webbpsf.utils.get_webbpsf_data_path(), 'NIRCam', 'OPD', 'wss_target_phase_fp6.fits'
+            )
+
+        # target phase map for prev. Needed just when the sensing aperture changed
+
+        if prev_sensing_apername == 'NRCA3_FP1':
+            prev_was_targ_file = os.path.join(
+                webbpsf.utils.get_webbpsf_data_path(), 'NIRCam', 'OPD', 'wss_target_phase_fp1.fits'
+            )
+        elif prev_sensing_apername == 'NRCA1_FP6':
+            prev_was_targ_file = os.path.join(
+                webbpsf.utils.get_webbpsf_data_path(), 'NIRCam', 'OPD', 'wss_target_phase_fp6.fits'
+            )
+
+
```

Comment:
So for instance this block would become just 
```
was_targ_file = utils.get_target_phase_map_filename(sensing_apername)
prev_was_targ_file = utils.get_target_phase_map_filename(prev_sensing_apername)
```

Reply:
done